.github/workflows/build.yml

```diff
@@ -1,56 +1,55 @@
-# This is a basic workflow to help you get started with Actions
+# build.yml
+# checks and test build on push and pull request to master branch
+# creates a pre-release on push
+name: build
 
-name: pre-release
-
-# Controls when the action will run. 
 on:
-  # Triggers the workflow on push but only for the master branch
   push:
-    branches: [ master ]
-
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
+    branches:
+      - "master"
+  pull_request:
+    branches:
+      - "master"
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  pre-release:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
-    steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - name: Check out repository code
-        uses: actions/checkout@v2
-
-      - name: Build
-        run: |
-          yarn install
-          npm run build
-          zip release/hypercrx.zip -r build/*
-
-      - name: Test output
-        shell: pwsh
-        run: Test-Path release/hypercrx.crx | Should -Be $true
-
-      - name: Release the extension
-        uses: "marvinpinto/action-automatic-releases@latest"
-        with:
-          repo_token: "${{ secrets.GITHUB_TOKEN }}"
-          automatic_release_tag: "latest"
-          prerelease: true
-          title: "Development Build"
-          files: |
-            release/hypercrx.crx
-            release/hypercrx.zip
-
-      - name: Update oss
-        uses: tvrcgo/upload-to-oss@v0.1.1
-        with:
-          key-id: ${{ secrets.accessKeyId }}
-          key-secret: ${{ secrets.accessKeySecret }}
-          region: ${{ secrets.region }}
-          bucket: ${{ secrets.bucket }}
-          asset-path: ./publish
-          target-path: ${{ secrets.deployPath }}
+  check-build-test:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Check out repository code
+          uses: actions/checkout@v2
+
+        - name: Setup Node.js
+          uses: actions/setup-node@v3
+          with:
+            cache: "yarn"
+
+        - name: install dependencies
+          run: |
+            yarn install
+
+        - name: Format checks
+          run: |
+            yarn run prettier:check
+
+        - name: Build
+          run: |
+            yarn run build
+
+        # https://github.com/marketplace/actions/file-existence
+        - name: Check file existence
+          id: check_files
+          uses: andstor/file-existence-action@v1
+          with:
+            files: "release/hypercrx.crx"
+            allow_failure: true
```

Comment:
Hey @wxharry, what does `allow_failure: true` do? Will it allow the workflow to continue even if the build fails?

Reply:
Just removing it is OK, I suppose. See [doc](https://github.com/andstor/file-existence-action#options-%EF%B8%8F).

Reply:
My intention was to fail action if file doesn't exist. We have an if branch to detect the output, so it is safe to remove it.

Reply:
In fact, on the contrary, setting `allow_failure` to `true` will prevent the action from being failed. 🤣

Reply:
My mistake. I just misunderstood its usage.